src/main/java/at/makubi/maven/plugin/avrohugger/GeneratorMojo.java

```diff
@@ -52,7 +58,20 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         String outputDirectoryPath = outputDirectory.getAbsolutePath();
         getLog().info("Generating Scala files for schemas in " + sourceDirectoryPath + " to " + outputDirectoryPath);
 
+        Map<String, String> nonNullNamespaceMapping = new HashMap<>();
+        if (namespaceMapping != null && !namespaceMapping.isEmpty()) {
```

Comment:
What do you think about using an empty `List` as default value? This would remove the `null` check.
You can remove the `if` statement completely as for empty `List`s the code would be executed zero times.

Please also move the transformation to `Map` to the `AvrohuggerGenerator` class since we need a `Scala` `Map` anyways.